gcloud/datastore/__init__.py

```diff
@@ -170,4 +182,7 @@ def allocate_ids(incomplete_key, num_ids):
     :rtype: list of :class:`gcloud.datastore.key.Key`
     :returns: The (complete) keys allocated with `incomplete_key` as root.
     """
-    return _require_dataset().allocate_ids(incomplete_key, num_ids)
+    dataset = _require_dataset()
+    connection = _require_connection()
+    return connection_module.allocate_ids(incomplete_key, num_ids,
```

Comment:
I'm happy you stuck here because it reminds of something I forgot to bring up.

How do you feel about having a single definition of `allocate_ids` just in `__init__`. We initially thought putting the methods from `Dataset` into `connection.py` as free functions, but why not just put them all in `__init__`?